src/lib.rs

```diff
@@ -292,10 +293,40 @@ impl<T: Config> Module<T> {
         }
 
         <EnclaveRegistry<T>>::remove(new_enclaves_count);
-
         Ok(())
     }
 
+    fn list_silent_workers(now: T::Moment) -> Vec<T::AccountId> {
+        let mut silent_workers: Vec<T::AccountId> = Vec::new();
+        let enclaves: Vec<(u64, Enclave<T::AccountId, Vec<u8>>)> =
+            <EnclaveRegistry<T>>::iter().collect();
+
+        let minimum = (now - T::SilenceDuration::get()).saturated_into::<u64>();
+
+        for enclave in enclaves {
+            if enclave.1.timestamp < minimum {
+                silent_workers.push(enclave.1.pubkey);
+            }
+        }
+        silent_workers
+    }
```

Comment:
This function could look a bit more ergonomic if we use some functional language features, i.e,

``` 
<EnclaveRegistry<T>>::iter()
.filter(|e| e.1.timestamp < minimum)
.map(|e| e.1.pubkey)
```

Furthermore, we only use it in line 316, where we turn the vector into an iterator again. So, we could spare the `collect()` at the end and return the iterator to safe computational power. However,  I don't know if `::iter()` returns references or owned values. If it is references, you might get some lifetime issues.

Then it might be easier to simply drop the function and just use these three lines in the function below in L314.

Reply:
Yes, much better

Reply:
Actually, I think we should rename the method on_timestamp_set to unregister_silent_workers. 
What about the generated RemovedEnclave? Shall we catch it somewhere? In thw worker?
What about an error, in case the enclave cannot be unregistered?

Reply:
I think you are right., we should rename the function to `unregister_silent_workers`.

I am not certain if we are listening for events currently in the worker. If yes, we can add this event. But as we currently don't cache the peers on the worker side, this is not really relevant.